diff --git a/src/useStore.ts b/src/useStore.ts
index 83440046da91b4efd45d83e6e4aacd60b995c1cc..ad1b81cdf5004159f2da0087166f6a59c63fd5b3 100644
--- a/src/useStore.ts
+++ b/src/useStore.ts
@@ -43,5 +43,10 @@ export function useStore<TSource, TSelected = NoInfer<TSource>>(
 			options: UseSelectorOptions<TSelected>,
 			slot?: symbol,
 		) => TSelected
-	)(source, selector, { compare }, slot);
+	)(
+		source,
+		selector,
+		compare === undefined ? {} : { compare },
+		slot,
+	);
 }
